### PR TITLE
migrate plugin to be compatible with ES 5.5.2

### PR DESCRIPTION
**Pull request by jcantrill**
This PR migrates the openshift-elasticsearch-plugin to be compatible with ES5.5.2

### CODE REVIEW

**Comment by jcantrill** [comment]
test

**Comment by jcantrill** [comment]
reset

**Comment by jcantrill** [comment]
[(5) build/port openshift-elasticsearch-plugin to 5.x [logging][epic-ois-es5]](https://trello.com/c/sWMbQwn8/542-5-build-port-openshift-elasticsearch-plugin-to-5x-loggingepic-ois-es5)

**Comment by ewolinetz** [comment]
@jcantrill so its captured in the pr... we're including the built jars for searchguard because we're making custom changes to it so that we can still have the same multitenant functionality?

File: src/main/java/io/fabric8/elasticsearch/plugin/filter/FieldStatsResponseFilter.java [hunk 1]
```diff
@@ -45,65 +44,62 @@
...
     
-    @Inject
     public FieldStatsResponseFilter(PluginClient client) {
         this.client = client;
     }
     
     @Override
     public int order() {
-        return Integer.MIN_VALUE;
+        return Integer.MAX_VALUE;
```
Comment (ewolinetz) [inline src/main/java/io/fabric8/elasticsearch/plugin/filter/FieldStatsResponseFilter.java]: This should run last now?

File: src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaSeed.java [hunk 2]
```diff
@@ -207,31 +197,29 @@ private boolean initialSeedKibanaIndex(final OpenshiftRequestContext context, Cl
...
-                esClient.admin().indices().create(createRequest).actionGet();
 
                 // Wait for health status of YELLOW
-                ClusterHealthRequest healthRequest = new ClusterHealthRequest().indices(new String[] { userIndex })
-                        .waitForYellowStatus();
-
-                esClient.admin().cluster().health(healthRequest).actionGet().getStatus();
+                esClient.admin().cluster().prepareHealth(userIndex)
+                    .setWaitForYellowStatus()
+                    .get()
```
Comment (ewolinetz) [inline src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaSeed.java]: .getStatus() and .get() aren't redundant?

File: src/main/java/io/fabric8/elasticsearch/util/RequestUtils.java [hunk 3]
```diff
@@ -16,140 +16,242 @@
...
+        return executePrivilegedAction(new PrivilegedAction<Boolean>(){
+
+            @Override
+            public Boolean run() {
+                final String user = StringUtils.defaultIfEmpty(getUser(request), "<UNKNOWN>");
+                final String token = getBearerToken(request);
+                boolean allowed = false;
+                Config config = new ConfigBuilder().withOauthToken(token).build();
+                try (NamespacedOpenShiftClient osClient = (NamespacedOpenShiftClient) k8ClientFactory.create(config)) {
+                    LOGGER.debug("Submitting a SAR to see if '{}' is able to retrieve logs across the cluster", user);
```
Comment (ewolinetz) [inline src/main/java/io/fabric8/elasticsearch/util/RequestUtils.java]: Remove?

File: src/main/java/io/fabric8/elasticsearch/util/RequestUtils.java [hunk 4]
```diff
@@ -16,140 +16,242 @@
...
+                    LOGGER.debug("Verifying user {} matches the given token.", user);
+                    Request okRequest = new Request.Builder()
+                            .addHeader(AUTHORIZATION_HEADER, "Bearer " + token)
+                            .url(osClient.getMasterUrl() + "oapi/v1/users/~")
+                            .build();
+                    Response response = null;
+                    try {
+                        response = osClient.getHttpClient().newCall(okRequest).execute();
+                        final String body = response.body().string();
+                        if (LOGGER.isDebugEnabled()) {
```
Comment (ewolinetz) [inline src/main/java/io/fabric8/elasticsearch/util/RequestUtils.java]: +1 to wrapping, we should probably wrap all the debug statements...

**Comment by jcantrill** [comment]
Closing as pushed a 5.5.x branch

**Pull request closed without merging**